--- init.c.old	2024-04-18 22:53:16.825867092 +0800
+++ init.c	2024-04-18 23:00:59.065761360 +0800
@@ -11,7 +11,6 @@
 //config:config INIT
 //config:	bool "init (10 kb)"
 //config:	default y
-//config:	select FEATURE_SYSLOG
 //config:	help
 //config:	init is the first program run when the system boots.
 //config:
@@ -128,7 +128,6 @@
 
 #include "libbb.h"
 #include "common_bufsiz.h"
-#include <syslog.h>
 #ifdef __linux__
 # include <linux/vt.h>
 # include <sys/sysinfo.h>